libs-scala/resources/src/test/suite/scala/com/digitalasset/resources/ResourceOwnerSpec.scala

```diff
@@ -783,6 +882,56 @@ final class ResourceOwnerSpec extends AsyncWordSpec with Matchers {
         succeed
       }
     }
+
+    "convert to a ResourceOwner, which does not ensure, that no TimerTask is running after the resource is released, if waitForRunningTasks = false" in {
+      val finishLongTakingTask = Promise[Unit]()
+      val timerResource =
+        Factories.forTimer(() => new Timer("test timer"), waitForRunningTasks = false).acquire()
+      for {
+        timer <- timerResource.asFuture
+        _ <- {
+          timer.schedule(
+            new TimerTask {
+              override def run(): Unit = {
+                Await.result(finishLongTakingTask.future, Duration(10, "seconds"))
+              }
+            },
+            0L,
+          )
+          info("As scheduled a long taking task")
+          val released = timerResource.release()
+          info("And as triggered release of the timer resource")
+          released
+        }
+      } yield {
+        info("Timer released")
+        Thread.sleep(100)
+        info("And as waiting 100 millis")
+        val extraTimerTaskStarted = Promise[Unit]()
+        timer.schedule(
+          new TimerTask {
+            override def run(): Unit = extraTimerTaskStarted.success(())
+          },
+          0L,
+        )
+        info("And scheduling a further task is still possible")
+        finishLongTakingTask.success(())
+        info("As completing the currently running timer task")
+        eventually {
+          an[IllegalStateException] should be thrownBy timer.schedule(
+            new TimerTask {
+              override def run(): Unit = ()
+            },
+            0,
+          )
+        }
+        info("Eventually scheduling new task on the released timer is not possible anymore")
+        Thread.sleep(100)
```

Comment:
1. Repetition of Thread.sleep(100) can be replaced by a `private def sleep()` 

2. I also wonder if it would not be better to use sth like `CountdownLatch` or Semaphores to `push` / `wait` steps instead of sleeps.

3. Also this last test is only  understandable thanks to info messages, maybe it would be good to part it into named methods/steps. Or use helper methods.

Reply:
1: I am not sure that helps us tremendously here
2: You are right, but I think this is good enough considering how simple it is. Please note: for the last two tests: those threads will be interrupted, so it is event driven despite the sleep
3: I prefer nice info messages which are visible in test runtime too, and follow natural text

Reply:
Also please note the very same content is already in main. This is a backport PR, if possible I would change the least here.